### PR TITLE
Reenabled registry dc's liveness probe

### LINKED ISSUE #5116: Reenable liveness probe in registry deployment config
Now that we're no longer trying to support the v1 registry, we should be able to reenable the liveness probe that `oadm registry` adds to the registry's DC:

https://github.com/openshift/origin/blob/3dd75654ae2cd6d4f762da95395d02ed496778a2/pkg/cmd/admin/registry/registry.go#L253-L265

@smarterclayton @mfojtik @miminar @pweil-

### PR DESCRIPTION
**Pull request by miminar**
Reverts 12f4a35e98f7a358a9763e06444083dddd3a29b0

Resolves #5116

### CODE REVIEW

**Comment by miminar** [comment]
@ncdc, @soltysh: WDYT?

**Comment by soltysh** [comment]
The changes LGTM, but I have one question. Are we going to base the liveness probe based on `HTTP/1.1 401 Unauthorized`, that's what I got when trying to access `/v2/` endpoint. Isn't there specifically `/healthz` endpoint we can access?

File: test/end-to-end/core.sh [hunk 1]
```diff
@@ -88,7 +88,7 @@ registry="$(dig @${API_HOST} "docker-registry.default.svc.cluster.local." +short
 [[ -n "${registry}" && "${registry}:5000" == "${DOCKER_REGISTRY}" ]]
 
 echo "[INFO] Verifying the docker-registry is up at ${DOCKER_REGISTRY}"
-wait_for_url_timed "http://${DOCKER_REGISTRY}/healthz" "[INFO] Docker registry says: " $((2*TIME_MIN))
+wait_for_url_timed "http://${DOCKER_REGISTRY}/v2/" "[INFO] Docker registry says: " $((2*TIME_MIN))
```
Comment (mfojtik) [inline test/end-to-end/core.sh]: will this return 401 to console?

**Comment by miminar** [comment]
@soltysh I made a wrong assumption based on unsecured upstream registry and didn't run tests to verify it. They [say](https://docs.docker.com/registry/deploying/#load-balancing-considerations):

> A properly secured registry should return 401 when the “/v2/” endpoint is hit without credentials. The response should include a WWW-Authenticate challenge, providing guidance on how to authenticate, such as with basic auth or a token service. If the load balancer has health checks, it is recommended to configure it to consider a 401 response as healthy and any other as down. This will secure your registry by ensuring that configuration problems with authentication don’t accidentally expose an unprotected registry. If you’re using a less sophisticated load balancer, such as Amazon’s Elastic Load Balancer, that doesn’t allow one to change the healthy response code, health checks can be directed at “/”, which will always return a 200 OK response.

`/` endpoint is not yet supported in `v2.0.1` registry. Therefor it should be correct to check `/v2/` which in our case returns 401. Unfortunately we can't use `probe.Handler.HTTPGet` action on it because the action considers endpoint ready if it returns `status >= 200 && status < 400`. So we would have to use an ExecAction like:

    /bin/bash -c 'curl -i -s 127.0.0.1:5000/v2/ | grep -q "HTTP/1\.[0-9]\s\+401\>"'

Which isn't much pretty. To sum up our options:

1. use `/v2/` option and expect 401 in probes and tests
2. stick with our custom `/healthz` route until the registry is rebased ... then switch to `/`
3. use `/v2` route which returns 301 which is considered *alive* by kubernetes probe handler
4. modify our [authorization check](https://github.com/openshift/origin/blob/master/pkg/dockerregistry/server/auth.go#L119) to return 200 for `/v2/` route

To avoid violating upstream's REST API, I'm now more inclined to stick with our `/healthz` route (opt. 2). WDYT?

**Comment by soltysh** [comment]
In my opinion we should wait then for the next rebase picking `/`. Testing 401 isn't good idea at the moment. We've waited this long, I think we should wait a few more days longer instead of introducing yet another anomalies. I'd rather see #5073 and #5404 in and then we can back to this.

**Comment by miminar** [comment]
@soltysh I agree.

**Comment by miminar** [comment]
I'll resubmit once the #5404 is resolved.

**Pull request closed without merging**